discussions/tasks.py

```diff
@@ -47,28 +78,44 @@ def sync_discussion_users():
 
 
 @app.task()
-def add_users_to_channel(channel_name, user_ids):
+def add_moderators_to_channel(channel_name):
     """
-    Add users to a open-discussions channel as contributors and subscribers
-
-    Args:
-        channel_name (str): The name of the channel
-        user_ids (list of int): profile ids to sync
+    Add moderators to a open-discussions channel
     """
     if not settings.FEATURES.get('OPEN_DISCUSSIONS_USER_SYNC', False):
         log.debug('OPEN_DISCUSSIONS_USER_SYNC is set to False (so disabled) in the settings')
         return
 
-    api.add_users_to_channel(channel_name, user_ids)
+    api.add_moderators_to_channel(channel_name)
 
 
 @app.task()
-def add_moderators_to_channel(channel_name):
+def sync_channel_memberships():
     """
-    Add moderators to a open-discussions chane
+    Syncs outstanding channel memberships
     """
     if not settings.FEATURES.get('OPEN_DISCUSSIONS_USER_SYNC', False):
         log.debug('OPEN_DISCUSSIONS_USER_SYNC is set to False (so disabled) in the settings')
         return
 
-    api.add_moderators_to_channel(channel_name)
+    # establish when we'll stop syncing this "batch"
+    end_time = now_in_utc() + timedelta(seconds=SYNC_MEMBERSHIPS_LOCK_TTL_SECONDS)
+    lock = _get_sync_memberships_lock()
+
+    def _get_memberships():
+        """Generator for membership ids to sync"""
+        for membership_id in api.get_membership_ids_needing_sync():
+            # this will stop yielding once our lock has expired
+            if end_time > now_in_utc():
+                yield membership_id
+            else:
+                break
```

Comment:
Can we add a test for this case? It would help to verify that the task will not run too long